test/e2e/cases/hcFromProbe.go

```diff
@@ -0,0 +1,101 @@
+package e2e
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/golang/glog"
+	kubeclient "k8s.io/client-go/kubernetes"
+
+	"github.com/GoogleCloudPlatform/k8s-multicluster-ingress/app/kubemci/pkg/gcp/cloudinterface"
+	utilsnamer "github.com/GoogleCloudPlatform/k8s-multicluster-ingress/app/kubemci/pkg/gcp/namer"
+)
+
+const (
+	lbNamePrefix = "mci1" // TODO get from loadbalancersyncer.go
+	svcNodePort = 30061
+	expectedHcPath = "/healthz"
+)
+
+func RunHCFromProbeTest() {
+	project, kubeConfigPath, lbName, ipName, clients := setupBasicDependencies()
+	setupHCFromProbe(kubeConfigPath, ipName, clients)
+	defer func() {
+		cleanupHCFromProbe(kubeConfigPath, ipName, clients)
+		cleanupBasicDependencies(ipName)
+	}()
+
+	testHCFromProbe(project, kubeConfigPath, lbName)
+}
+
+func setupHCFromProbe(kubeConfigPath, ipName string, clients map[string]kubeclient.Interface) {
+	args := []string{"kubectl", fmt.Sprintf("--kubeconfig=%s", kubeConfigPath), "-f", "testdata/e2e/hcFromProbe/app/"}
+	for k := range clients {
+		glog.Infof("Creating app in cluster %s", k)
+		createArgs := append(args, []string{"create", fmt.Sprintf("--context=%s", k)}...)
+		// kubectl create may fail if this was setup in a previous run.
+		runCommand(createArgs)
+	}
+	if _, err := runCommand([]string{"sed", "-i", "-e", fmt.Sprintf("s/\\$E2E_KUBEMCI_IP/%s/", ipName), "testdata/e2e/hcFromProbe/ingress.yaml"}); err != nil {
+		glog.Fatalf("Error updating http ingress yaml with sed: %v", err)
+	}
+}
+
+func cleanupHCFromProbe(kubeConfigPath, ipName string, clients map[string]kubeclient.Interface) {
+	args := []string{"kubectl", fmt.Sprintf("--kubeconfig=%s", kubeConfigPath), "-f", "testdata/e2e/hcFromProbe/app/"}
+	for k := range clients {
+		glog.Infof("Deleting app from cluster %s", k)
+		deleteArgs := append(args, []string{"delete", fmt.Sprintf("--context=%s", k)}...)
+		runCommand(deleteArgs)
+	}
+	runCommand([]string{"sed", "-i", "-e", fmt.Sprintf("s/%s/\\$E2E_KUBEMCI_IP/", ipName), "testdata/e2e/hcFromProbe/ingress.yaml"})
+}
+
+func testHCFromProbe(project, kubeConfigPath, lbName string) {
```

Comment:
Most of this is just copied from the basic test, I could try to break it out into a reusable method.

Reply:
yes that will be great :)